app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitSponsorApiController.php

```diff
@@ -983,4 +993,167 @@ public function deleteSocialNetwork($summit_id, $sponsor_id, $social_network_id)
 
         });
     }
+
+    // Extra Questions
+
+    /**
+     * @param $summit_id
+     * @param $sponsor_id
+     * @return \Illuminate\Http\JsonResponse|mixed
+     */
+    public function getExtraQuestions($summit_id, $sponsor_id)
+    {
+        $summit = SummitFinderStrategyFactory::build($this->getSummitRepository(), $this->resource_server_context)->find($summit_id);
+        if (is_null($summit)) return $this->error404();
+
+        $sponsor = $summit->getSummitSponsorById(intval($sponsor_id));
+        if (is_null($sponsor)) return $this->error404();
+
+        return $this->_getAll(
+            function () {
+                return [
+                    'name' => ['==', '=@'],
+                    'label' => ['==', '=@'],
+                    'class' => ['==']
+                ];
+            },
+            function () {
+                return [
+                    'name' => 'sometimes|string',
+                    'label' => 'sometimes|string',
+                    'class' => 'sometimes|string|in:'.join(",", ExtraQuestionTypeConstants::AllowedQuestionClass)
```

Comment:
this restriction of class is too wide please check SDS requirements

Reply:
It is exactly what the filter mapping in the repository expects.